### PR TITLE
[scheduler] Replace scheduler-extender

### PR DESCRIPTION
**Pull request by Ranger-X**
## Description
Make linstor-scheduler-extender to use Deckhouse's `KubeSchedulerWebhookConfiguration`.

## Why do we need it, and what problem does it solve?
Use a centralized way to connect scheduler extenders in Deckhouse's modules.

## What is the expected result?
Users should NOT notice any changes to their clusters.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

File: images/linstor-scheduler-extender/patches/stork/stork.patch [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+ 	decoder := json.NewDecoder(req.Body)
+ 	defer func() {
+ 		if err := req.Body.Close(); err != nil {
+@@ -164,6 +200,13 @@ func (e *Extender) processFilterRequest(w http.ResponseWriter, req *http.Request
+ 		return
+ 	}
+
++	args, err = fillNodesByNames(args)
++	if err != nil {
++		log.Errorf("Cannot fill nodes: %+v", err)
```
Comment (astef) [inline images/linstor-scheduler-extender/patches/stork/stork.patch:79]: Here, and in other places, `%v` is good enough format in order to use output of `err.Error()`. Getting that string by yourself and rendering it with `%s` is even better.

File: images/linstor-scheduler-extender/patches/stork/stork.patch [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
++// added by Flant.com
++func fillNodesByNames(inputData schedulerapi.ExtenderArgs) (schedulerapi.ExtenderArgs, error) {
++	if inputData.NodeNames != nil && len(*inputData.NodeNames) > 0 {
++		if inputData.Nodes != nil && len(inputData.Nodes.Items) > 0 {
++			// nodes already presents in inputData
++			return inputData, nil
++		}
++	} else {
++		// return same nodes in case there is no NodeNames present
++		return inputData, fmt.Errorf("no NodeNames")
```
Comment (astef) [inline images/linstor-scheduler-extender/patches/stork/stork.patch:117]: Don't return results together with `err`, unless there's a strong reason to.

File: images/linstor-scheduler-extender/patches/stork/stork.patch [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
++		err := error(nil)
++		tlsMode := len(e.CertFile) > 0 && len(e.KeyFile) > 0
++
++		if tlsMode {
++			if _, exist := os.Stat(e.CertFile); errors.Is(exist, os.ErrNotExist) {
++				log.Warnf("CertFile `%s` does not exists. Launch as HTTP", e.CertFile)
++				tlsMode = false
++			}
++
++			if _, exist := os.Stat(e.KeyFile); errors.Is(exist, os.ErrNotExist) {
```
Comment (astef) [inline images/linstor-scheduler-extender/patches/stork/stork.patch:40]: `exist` is not a bool flag, but an error, so it's better to call it `err`

File: images/linstor-scheduler-extender/patches/stork/stork.patch [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
++
++// block added by flant.com
++func fillNodesByNames(inputData *schedulerapi.ExtenderArgs) (error) {
++	if inputData.NodeNames != nil && len(*inputData.NodeNames) > 0 {
++		if inputData.Nodes != nil && len(inputData.Nodes.Items) > 0 {
++			// nodes already presents in inputData, just return 'as is'
++			return nil
++		}
++	} else {
++		return fmt.Errorf("no NodeNames")
```
Comment (Ranger-X) [inline images/linstor-scheduler-extender/patches/stork/stork.patch:248]: check for .Nodes existense

File: templates/linstor-scheduler/configmap.yaml [hunk 5]
```diff
@@ -16,10 +17,13 @@ data:
...
-      - urlPrefix: http://localhost:8099
+      - urlPrefix: https://localhost:8099
         filterVerb: filter
         prioritizeVerb: prioritize
         weight: 5
-        enableHTTPS: false
-        httpTimeout: 300000s
+        enableHTTPS: true
+        httpTimeout: 300s
         nodeCacheCapable: false
```
Comment (Ranger-X) [inline templates/linstor-scheduler/configmap.yaml:26]: set to true

**Review by AleksZimin — DISMISSED** [review]

**Review by astef — APPROVED** [review]

**Review by AleksZimin — APPROVED** [review]

**Pull request merged**